packages/clay-core/src/vertical-nav/VerticalNav.tsx

```diff
@@ -242,53 +248,86 @@ function VerticalNav<T extends Record<string, any> | string>({
 		<nav
 			className={classNames('menubar menubar-transparent', {
 				['menubar-decorated']: decorated,
+				['menubar-primary']: displayType === 'primary',
 				['menubar-vertical-expand-lg']: large,
-				['menubar-vertical-expand-md']: !large,
+				['menubar-vertical-expand-md']:
+					!large && displayType !== 'primary',
 			})}
 		>
-			<CustomTrigger onClick={() => setIsOpen(!isOpen)}>
-				<span className="inline-item inline-item-before">
-					{triggerLabel}
-				</span>
-
-				<Icon
-					focusable="false"
-					role="presentation"
-					spritemap={spritemap}
-					symbol="caret-bottom"
-				/>
-			</CustomTrigger>
-
-			<div
-				{...navigationProps}
-				className={classNames('collapse menubar-collapse', {
-					show: isOpen,
-				})}
-				ref={containerRef}
-			>
-				<Nav aria-orientation="vertical" nested role="menubar">
-					<VerticalNavContext.Provider
-						value={{
-							activeKey:
-								active && collection.hasItem(active)
-									? active
-									: hasDepthActive
-									? null
-									: undefined,
-							ariaCurrent: ariaCurrent ? 'page' : null,
-							childrenRoot: childrenRootRef,
-							close,
-							expandedKeys,
-							firstKey: collection.getFirstItem().key,
-							open,
-							spritemap,
-							toggle,
-						}}
+			{displayType !== 'primary' && (
+				<>
+					<CustomTrigger onClick={() => setIsOpen(!isOpen)}>
+						<span className="inline-item inline-item-before">
+							{triggerLabel}
+						</span>
+
+						<Icon
+							focusable="false"
+							role="presentation"
+							spritemap={spritemap}
+							symbol="caret-bottom"
+						/>
+					</CustomTrigger>
+
+					<div
+						{...navigationProps}
+						className={classNames('collapse menubar-collapse', {
+							show: isOpen,
+						})}
+						ref={containerRef}
 					>
-						<Collection<T> collection={collection} />
-					</VerticalNavContext.Provider>
-				</Nav>
-			</div>
+						<Nav aria-orientation="vertical" nested role="menubar">
+							<VerticalNavContext.Provider
+								value={{
+									activeKey:
+										active && collection.hasItem(active)
+											? active
+											: hasDepthActive
+											? null
+											: undefined,
+									ariaCurrent: ariaCurrent ? 'page' : null,
+									childrenRoot: childrenRootRef,
+									close,
+									expandedKeys,
+									firstKey: collection.getFirstItem().key,
+									open,
+									spritemap,
+									toggle,
+								}}
+							>
+								<Collection<T> collection={collection} />
+							</VerticalNavContext.Provider>
+						</Nav>
```

Comment:
You could store this in a constant to avoid repeating the code and reuse it below.

Reply:
Thanks @matuzalemsteles . I tried my best to put it in a `const`. I'm having trouble figuring out typescript after moving it.

Reply:
Hmm let me see, I'll push a commit dealing with that.